File: android_okmock/okmock-interceptor/src/main/java/nolambda/gadb/okmock/adapter/OkMockPayload.kt
package nolambda.gadb.okmock.adapter

data class OkMockPayload(
    val matcher: Matcher,
    val mock: Mock
)

data class Matcher(
    val path: Regex,
    val method: String
)

data class Mock(
    val code: Int,
    val body: String,
    val message: String,
    val headers: Map<String, String>
)

File: flutter_okmock/sample/lib/OkMockPayload.dart
class OkMockPayload {
  String path;
  String body;
  String method;
  int code;
  String message;
  Map<String, String> headers;

  OkMockPayload.fromJson(Map<String, dynamic> json) {
    path = json["path"];
    body = json["body"];
    method = json["method"];
    code = json["code"];
    message = json["message"];
    headers = json["headers"];
  }
}
